### PR TITLE
Add options bag to `addTransaction` public API method

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

In order to align Tx controller public APIs between mobile & extension, this PR aims to add an options bag instead of inline arguments for `addTransaction` function. 

### `@metamask/transaction-controller`

- **BREAKING**: Adds option bag to `addTransaction` public API method instead of inline arguments.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.ts [hunk 1]
```diff
@@ -335,14 +347,12 @@ export class TransactionController extends BaseController<
...
-   * @param origin - The domain origin to append to the generated TransactionMeta.
-   * @param deviceConfirmedOn - An enum to indicate what device the transaction was confirmed to append to the generated TransactionMeta.
+   * @param opts - The options bag for addTransaction.
    * @returns Object containing a promise resolving to the transaction hash if approved.
    */
   async addTransaction(
     transaction: Transaction,
-    origin?: string,
-    deviceConfirmedOn?: WalletDevice,
+    opts: AddTransactionOptions = {},
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:355]: Do we need a separate type here as it won't be used anywhere else? Plus we could inline all the variables and dereference them so we can use them as if they were standard arguments?

As done in the extension for reference.

File: packages/transaction-controller/src/TransactionController.ts [hunk 2]
```diff
@@ -124,6 +124,18 @@ export interface TransactionState extends BaseState {
...
 
+/**
+ * @type AddTransactionOptions
+ *
+ * Transaction controller addTransaction method options
+ * @param origin - The domain origin to append to the generated TransactionMeta.
+ * @param deviceConfirmedOn - An enum to indicate what device the transaction was confirmed to append to the generated TransactionMeta.
+ */
+type AddTransactionOptions = {
+  origin?: string;
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:135]: Minor, but as this list will likely grow, can we keep them in alphabetical order?

File: packages/transaction-controller/src/TransactionController.ts
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:1]: It's a little worrying we didn't need to change any of the unit tests for this.

Should we add some small unit tests to verify we support these arguments and assert they are stored in the transaction metadata?

File: packages/transaction-controller/src/TransactionController.ts [hunk 3]
```diff
@@ -124,6 +124,18 @@ export interface TransactionState extends BaseState {
   methodData: { [key: string]: MethodData };
 }
 
+/**
+ * @type AddTransactionOptions
+ *
+ * Transaction controller addTransaction method options
+ * @param origin - The domain origin to append to the generated TransactionMeta.
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts:131]: Removed

File: packages/transaction-controller/src/TransactionController.ts (hunk 2 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts:135]: Done

File: packages/transaction-controller/src/TransactionController.ts (hunk 1 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts:355]: Removed type, and converted to inline variables

File: packages/transaction-controller/src/TransactionController.ts
Comment (OGPoyraz) [reply packages/transaction-controller/src/TransactionController.ts:1]: For sure, I've added a small verification 👍

File: packages/transaction-controller/src/TransactionController.ts [hunk 4]
```diff
@@ -335,14 +335,20 @@ export class TransactionController extends BaseController<
    * if not provided. If A `<tx.id>:unapproved` hub event will be emitted once added.
    *
    * @param transaction - The transaction object to add.
-   * @param origin - The domain origin to append to the generated TransactionMeta.
-   * @param deviceConfirmedOn - An enum to indicate what device the transaction was confirmed to append to the generated TransactionMeta.
+   * @param opts - The options bag for addTransaction
+   * @param opts.deviceConfirmedOn - An enum to indicate what device the transaction was confirmed.
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:339]:
```suggestion
   * @param opts.deviceConfirmedOn - An enum to indicate what device confirmed the transaction.
```

File: packages/transaction-controller/src/TransactionController.ts [hunk 5]
```diff
@@ -335,14 +335,20 @@ export class TransactionController extends BaseController<
    * if not provided. If A `<tx.id>:unapproved` hub event will be emitted once added.
    *
    * @param transaction - The transaction object to add.
-   * @param origin - The domain origin to append to the generated TransactionMeta.
-   * @param deviceConfirmedOn - An enum to indicate what device the transaction was confirmed to append to the generated TransactionMeta.
+   * @param opts - The options bag for addTransaction
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:338]:
```suggestion
   * @param opts - Additional options to control how the transaction is added.
```

File: packages/transaction-controller/src/TransactionController.ts [hunk 6]
```diff
@@ -335,14 +335,20 @@ export class TransactionController extends BaseController<
...
+   * @param opts.deviceConfirmedOn - An enum to indicate what device the transaction was confirmed.
+   * @param opts.origin - The domain origin
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/TransactionController.ts:340]:
```suggestion
   * @param opts.origin - The origin of the transaction request, such as a dApp hostname.
```

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**